src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3595,6 +3784,97 @@ export class StudyViewPageStore {
         default: [],
     });
 
+    readonly genericAssayEntitiesGroupByGenericAssayType = remoteData<{
+        [genericAssayType: string]: GenericAssayMeta[];
+    }>({
+        await: () => [this.molecularProfiles],
+        invoke: async () => {
+            return await fetchGenericAssayMetaByMolecularProfileIdsGroupByGenericAssayType(
+                this.molecularProfiles.result
+            );
+        },
+    });
+
+    readonly genericAssayStableIdToMeta = remoteData<{
+        [genericAssayStableId: string]: GenericAssayMeta;
+    }>({
+        await: () => [this.genericAssayEntitiesGroupByGenericAssayType],
+        invoke: () => {
+            return Promise.resolve(
+                _.chain(this.genericAssayEntitiesGroupByGenericAssayType.result)
+                    .values()
+                    .flatten()
+                    .uniqBy(meta => meta.stableId)
+                    .keyBy(meta => meta.stableId)
+                    .value()
+            );
+        },
+    });
+
+    readonly genericAssayProfileOptionsByType = remoteData(
+        {
+            await: () => [
+                this.molecularProfiles,
+                this.sampleUniqueKeysByMolecularProfileIdSet,
+            ],
+            invoke: async () => {
+                return Promise.resolve(
+                    _.chain(this.molecularProfiles.result)
+                        .filter(
+                            profile =>
+                                profile.molecularAlterationType ===
+                                AlterationTypeConstants.GENERIC_ASSAY
+                        )
+                        .groupBy(profile => profile.genericAssayType)
+                        .mapValues(profiles =>
+                            getMolecularProfileOptions(
+                                profiles,
+                                this.sampleUniqueKeysByMolecularProfileIdSet
+                                    .result,
+                                undefined,
+                                true
+                            )
+                        )
+                        .value()
+                );
+            },
+            default: {},
+        },
+        {}
+    );
+
+    readonly genericAssayEntityOptionsByType = remoteData(
```

Comment:
looks similar to `genericAssayProfileOptionsByType` (line 3814)

Reply:
Oh thanks, this is an outdated function, it shouldn't be here, deleted.